ourhexenv.py

```diff
@@ -204,13 +205,16 @@ def step(self, action):
                 self.terminations = {agent: False for agent in self.agents}
 
         if self.agent_selection == "player_2":
-            # Pie rule should only be usable on the first move of Player 2
-            self.is_pie_rule_usable = True
+            # Player 2 has made their first move, make pie rule unusable
+            self.is_pie_rule_usable = False
 
         for agent in self.agents:
             self._cumulative_rewards[agent] += self.rewards[agent]
 
-        self.agent_selection = self.agent_selector.next()
+        #Advance to the next agent if the game is not over
+        if not any(self.terminations.values()):
```

Comment:
Please also check 
`self.truncations.values()`
When playing on massive boards we may wish for our agents to stop playing after a certain number of moves.

Reply:
This is out of scope, not necessary